src/main/java/com/oneops/secrets/command/SecretDelete.java

```diff
@@ -39,15 +40,24 @@
 
   @Override
   public void exec() {
-
+    Pattern p = Pattern.compile("^y(es)?$");
     try {
-      String in =
-          System.console()
-              .readLine(
-                  warn(
-                      "The delete secret operation is irrevocable. Do you want to proceed (y/n)? "));
-      if (in == null || !in.equalsIgnoreCase("y")) {
-        throw new IllegalStateException("Exiting");
+      if (null != System.console()) {
+        String in =
+            ConsoleImpl.readConsole(
+                "The delete secret operation is irrevocable. Do you want to proceed (y/n)? ");
+        if (in == null || !p.matcher(in.toLowerCase()).lookingAt()) {
+          throw new IllegalStateException("Exiting");
+        }
+      } else {
+        try {
+          String input = CommandUtil.readFromSystemIn(p);
+          if (input == null || !p.matcher(input.toLowerCase()).lookingAt()) {
```

Comment:
Why we need to use `lookingAt()` method here as `it does not require that the entire region be matched` ?